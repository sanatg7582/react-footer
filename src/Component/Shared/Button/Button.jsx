import React from "react";
import PropTypes from "prop-types";
import styles from "./Button.module.scss";

/**
 * function for Button anywhare and anytype
 * @param {string} buttonType // type of button
 * @param {string} buttonName // name of button
 * @param {string} btnClass // class of button
 * @param {Element} children // childern of button like icon
 * @param {func} handleOnClick // click function
 * @param {string} index // object index
 * @param {bool} status //
 * return button element
 */
const Button = props => {
  const {
    buttonType,
    buttonName,
    btnClass,
    children,
    handleOnClick,
    index,
    status
  } = props;
  return (
    <button
      type={buttonType}
      className={`${styles.btn} ${styles[btnClass]}`}
      onClick={e => handleOnClick(e, index, status)}
    >
      {buttonName ? buttonName : children}
    </button>
  );
};
// default props
Button.defaultProps = {
  buttonName: "",
  btnClass: "",
  children: <></>,
  handleOnClick: () => {},
  index: "",
  status: true
};
// proptypes validation
Button.propTypes = {
  buttonType: PropTypes.string.isRequired,
  buttonName: PropTypes.string,
  btnClass: PropTypes.string,
  children: PropTypes.element,
  handleOnClick: PropTypes.func,
  status: PropTypes.bool
};
export default Button;
