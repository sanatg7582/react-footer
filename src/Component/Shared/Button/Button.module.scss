.btn {
  color: #fff;
  border: none;
  padding: 8px 15px 8px 15px;
  border-radius: 5px;
  margin: 10px 0px 0px 0px;
  font-weight: 700;
}

.success {
  background-color: green;
}
.warning {
  background-color: orange;
}
.danger {
  background-color: red;
}
.primary {
  color: black;
}
