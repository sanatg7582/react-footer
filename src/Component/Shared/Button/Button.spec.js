import React from "react";
import { mount } from "enzyme";
import Button from "./Button";

describe("Load <Button />", () => {
  let wrapper = "";
  const handleOnClickEvent = jest.fn();
  beforeAll(() => {
    const props = {
      buttonType: "button",
      buttonName: "save",
      btnClass: "success",
      children: <></>,
      handleOnClick: handleOnClickEvent,
      index: 0,
      status: false
    };
    wrapper = mount(<Button {...props} />);
    expect(wrapper).toBeTruthy();
  });
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Button length should be 1", () => {
    expect(wrapper.find("button")).toHaveLength(1);
  });
  it("checking props", () => {
    expect(wrapper.find("button").props().type).toEqual("button");
    expect(wrapper.find("button").hasClass("btn success")).toEqual(true);
    expect(wrapper.find("button").type()).toEqual("button");
  });
  it("checking button click", () => {
    wrapper.find("button").simulate("click");
    expect(handleOnClickEvent).toHaveBeenCalled();
  });
  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
