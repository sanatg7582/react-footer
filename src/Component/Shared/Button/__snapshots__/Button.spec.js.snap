// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <Button /> SnapShot test 1`] = `ReactWrapper {}`;
